sdk/src/transaction.rs

```diff
@@ -255,6 +255,61 @@ impl Transaction {
         }
     }
 
+    /// Check keys and keypair lengths, then sign this transaction, using an assortment of
+    /// KeypairUtil trait objects
+    pub fn sign_dynamic_signers(
+        &mut self,
+        keypairs: &[&dyn KeypairUtil],
```

Comment:
This method looks to mix two independents concepts, fallible implementations, and heterogeneous lists. But it's not clear to me how this implements heterogeneous lists with a Box.

Reply:
I don't understand your comment here.

Just realizing that if we are able to make #8342 work, we will still need these fallible implementations.

Reply:
I mistyped. Should be "without a Box"

Reply:
I'm still confused. `&dyn Trait` is still a trait object, just the reference type.